src/modules/tests/test-data/cat-adi-part2/manoeuvres/manoeuvres.reducer.ts

```diff
@@ -1,14 +1,88 @@
 import { CatADI2UniqueTypes } from '@dvsa/mes-test-schema/categories/ADI2';
-import * as manoeuvresCatADI2Actions from './manoeuvres.actions';
+import * as manoeuvresActions from './manoeuvres.actions';
+import { CompetencyOutcome } from '../../../../../shared/models/competency-outcome';
 
-export const initialState: CatADI2UniqueTypes.Manoeuvres[] = [];
+export const initialState: CatADI2UniqueTypes.Manoeuvres[] = [{}, {}];
+
+let newState = initialState;
 
-// TODO - ADI2: impliment manoeuvres reducer
 export function manoeuvresCatADI2Reducer(
   state = initialState,
-  action: manoeuvresCatADI2Actions.Types,
+  action: manoeuvresActions.Types,
 ): CatADI2UniqueTypes.Manoeuvres[] {
   switch (action.type) {
+    case manoeuvresActions.RECORD_MANOEUVRES_SELECTION:
+      newState = state.slice(0);
+
+      newState[action.index] = {
+        [action.manoeuvre]: {
+          ...state[action.index][action.manoeuvre],
+          selected: true,
+        },
+      };
+
+      if (
+        action.index === 0 &&
+        state[1][action.manoeuvre] &&
+        state[1][action.manoeuvre].selected
+      ) {
+        newState[1][action.manoeuvre] = {};
+      }
+
+      return newState;
+    case manoeuvresActions.ADD_MANOEUVRE_DRIVING_FAULT:
+      newState = state.slice(0);
```

Comment:
also, would be good to not slice the state here

Reply:
Why not?

Reply:
Because it's too much logic and to be avoided in a reducer, conventions around the redux pattern.

The issue that you're trying to solve is very similar to the vehicle checks. Implementing the manoeuvre state with indexes would be the consistent, right solution.

Reply:
It is implemented with indexes. `slice` is just being used to return a fresh copy of the manoeuvres collection and not a reference.

Reply:
Reducers rewritten in a style which I hope is more familiar.